### PR TITLE
Global Footer

### PR DESCRIPTION
**Pull request by Krank3n**

### CODE REVIEW

File: src/scss/_theSupporters.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.supporters {
+  max-width: 1100px;
```
Comment (jakxnz) [inline src/scss/_theSupporters.scss:2]: What determines this `1100px`? Is it needed? Or should we be following our grid system?

File: src/js/components/Supporters/TheSupporters.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import React from 'react';
+import SupportService from '../config/supporter-config';
+
+const TheSupporters = () => (
+  <div className="support-banner p-0 h-auto">
+    <div className="supporters w-100 p-4 align-items-center d-flex flex-wrap">
+    {SupportService
+      .map((item) => (
+        <div className="m-auto p-3">
+          <a href={item.path}>
```
Comment (jakxnz) [inline src/js/components/Supporters/TheSupporters.js:11]: Do these also need `noreferrer`?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      <div className="p-4">
+        <a href="https://thomasandrewhansen.com" target="_blank" rel="noopener noreferrer">
+          <span className="mr-1 font-weight-bold">
+            J Darlow
+          </span>
+        </a>
+      </div>
+      <div>
+        <CImg
+          src="https://media-exp1.licdn.com/dms/image/C5603AQFCq5jKlKuFOQ/profile-displayphoto-shrink_200_200/0/1516800185433?e=1620259200&v=beta&t=TEizzz94vIHLCf0t6cUHhIEF10iBBjc-DNAM_7FHsBk"
```
Comment (jakxnz) [inline src/js/components/UnderflipTeam/UnderFlipTeam.js:35]: Could we just use the link to our avatars on github as the src?

File: src/js/components/Footer/TheFooter.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      <span className="mr-1">Developed by</span>
+      <a href="https://coreui.io/react" target="_blank" rel="noopener noreferrer">UnderFlip</a>
+    </div>
+    <div className="d-flex w-100 flex-wrap">
+      <UnderFlipTeam />
+    </div>
+  </CFooter>
+);
+
+export default React.memo(TheFooter);
```
Comment (jakxnz) [inline src/js/components/Footer/TheFooter.js:19]: Are we doing to bother memoizing stuff? I thought we were gonna go static (i.e NextJS)?

**Review by jakxnz — CHANGES_REQUESTED** [review]
Feels like you put this together pretty easily 😃. I've added question

File: src/js/components/Footer/TheFooter.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { CFooter, CImg } from '@coreui/react';
+import UnderFlipTeam from '../UnderflipTeam/UnderFlipTeam';
+
+const TheFooter = () => (
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/Footer/TheFooter.js:5]: Just based it off how COREUI had it. Didnt put thought into it. I'll change it to Footer.

File: src/js/components/Footer/TheFooter.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  <CFooter fixed={false} className="p-4 h-auto footers-banner">
+    <div className="m-auto">
+      <a className="mr-1" href="https://shredindex.com" target="_blank" rel="noopener noreferrer">ShredIndex</a>
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/Footer/TheFooter.js:8]: Sure, hope I have done it right.

File: src/js/components/Footer/TheFooter.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+      <span className="mr-1">&copy; 2020</span>
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/Footer/TheFooter.js:9]: sure

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { CImg } from '@coreui/react';
+import React from 'react';
+
+const UnderFlipTeam = () => (
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/UnderflipTeam/UnderFlipTeam.js:4]: sure

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        />
+      </div>
+      <div className="p-4">
+        <span className="mr-1 font-weight-bold">
+          <a href="https://thomasandrewhansen.com" target="_blank" rel="noopener noreferrer">
+            T
+            Hansen
+          </a>
+        </span>
+      </div>
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/UnderflipTeam/UnderFlipTeam.js:23]: sure, only issue with this is, The design is inverted. Which is minor, but cant think of best way to implement. Would I add a style to the data? Or base it off the index of the mapping, to flip the style?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+const UnderFlipTeam = () => (
+  <>
+    <div className="m-auto p-4 d-flex align-items-center">
+      <div>
+        <CImg
+          src="https://shredindex.com/storage/app/media/profile-tom.png"
+          className="c-avatar-img developer-avatar round"
+          width="120"
+          height="120"
+          alt="admin@bootstrapmaster.com"
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/UnderflipTeam/UnderFlipTeam.js:13]: cool

File: src/js/components/Supporters/TheSupporters.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+            <CImg
+              src={item.imgSrc}
+              className="supporters-img opacity-3"
+              alt={item.name}
+            />
+          </a>
+        </div>
+      ))}
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/Supporters/TheSupporters.js:19]: ahhhh yes :D

File: src/scss/_theSupporters.scss (hunk 1 above)
Comment (Krank3n) [reply src/scss/_theSupporters.scss:2]: Not needed, just looks better, I am trying to think if there is a way to add this using our current grid system... any ideas?

File: src/js/components/Supporters/TheSupporters.js (hunk 2 above)
Comment (Krank3n) [reply src/js/components/Supporters/TheSupporters.js:11]: taken out

File: src/js/components/Footer/TheFooter.js (hunk 4 above)
Comment (Krank3n) [reply src/js/components/Footer/TheFooter.js:19]: Taken Out

File: src/js/components/UnderflipTeam/UnderFlipTeam.js (hunk 3 above)
Comment (Krank3n) [reply src/js/components/UnderflipTeam/UnderFlipTeam.js:35]: can I flip your photo around so we are looking opposite directs lol?

File: src/js/components/SupportItems/SupportItems.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import { CImg } from '@coreui/react';
+import React from 'react';
+import SupportService from '../config/supporter-config';
+
+const SupportItems = () => (
+  <>
+    {SupportService
```
Comment (jakxnz) [inline src/js/components/SupportItems/SupportItems.js:7]: Can this be simplified?

File: src/js/components/SupportItems/SupportItems.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+      .map((item) => (
+        <div className="m-auto p-3">
+          <a href={item.path}>
```
Comment (jakxnz) [inline src/js/components/SupportItems/SupportItems.js:10]: Can we add `noopener noreferrer` here?

File: src/js/components/TeamMember/TeamMember.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
+import { CImg } from '@coreui/react';
+import React from 'react';
+
+const TeamMember = (props) => (
+  <>
```
Comment (jakxnz) [inline src/js/components/TeamMember/TeamMember.js:5]: Does this need the `<>` fragment?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import TeamConfig from '../config/team-config';
+import TeamMember from '../TeamMember/TeamMember';
+
+const UnderFlipTeam = () => (
+  <>
+    {TeamConfig
```
Comment (jakxnz) [inline src/js/components/UnderflipTeam/UnderFlipTeam.js:7]: Can this be simplified?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 16]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import TeamConfig from '../config/team-config';
+import TeamMember from '../TeamMember/TeamMember';
+
+const UnderFlipTeam = () => (
```
Comment (jakxnz) [inline src/js/components/UnderflipTeam/UnderFlipTeam.js:5]: "Underflip" is one word right? So is this component's name `UnderflipTeam`?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+  <>
+    {TeamConfig
+      .map((item) => (
+        <TeamMember member={item} />
+      ))}
+  </>
+);
```
Comment (jakxnz) [inline src/js/components/UnderflipTeam/UnderFlipTeam.js:12]:
```suggestion
const UnderFlipTeam = () => TeamConfig.map((item) => <TeamMember member={item} />);
```

File: src/js/components/config/team-config.js [hunk 18]
```diff
@@ -0,0 +1,14 @@
+const TeamConfig = [
+  {
+    name: 'T Hansen',
+    path: 'https://thomasandrewhansen.com',
+    imgSrc: 'https://avatars.githubusercontent.com/u/45204791?s=56&v=4',
+  },
+  {
+    name: 'J Darlow',
+    path: 'https://www.producthunt.com/posts/shred-index',
+    imgSrc: 'https://avatars.githubusercontent.com/u/1634995?s=60&u=ebaca76ccf999d03b9259721d800571f6df7a6a4&v=4',
```
Comment (jakxnz) [inline src/js/components/config/team-config.js:10]:
```suggestion
    name: 'jakxnz',
    path: 'https://github.com/jakxnz',
    imgSrc: 'https://avatars.githubusercontent.com/u/1634995?s=60&u=ebaca76ccf999d03b9259721d800571f6df7a6a4&v=4',
```

I'm not picky, but while I'm here 😂

File: src/scss/_theSupporters.scss (hunk 1 above)
Comment (jakxnz) [reply src/scss/_theSupporters.scss:2]: As in, bootstrap? `class="container"`?

File: src/scss/style.scss [hunk 19]
```diff
@@ -6,3 +6,8 @@
 
 // If you want to add something do it here
 @import "custom";
+
+// Component styles
+@import "theSupportersColor";
+@import "theSupporters";
+@import "theFooter";
```
Comment (jakxnz) [inline src/scss/style.scss:13]: Do we need the `the` prefix?

File: src/js/components/TeamMember/TeamMember.js [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+          src={props.member.imgSrc}
+          className="c-avatar-img developer-avatar round"
+          width="120"
+          height="120"
+          role="presentation"
+        />
+      </div>
+      <div className="p-4">
+        <span className="mr-1 font-weight-bold">
+          <a href={props.member.path} target="_blank" rel="noopener">
```
Comment (jakxnz) [inline src/js/components/TeamMember/TeamMember.js:18]:
```suggestion
          <a href={props.member.path} target="_blank" rel="noopener noreferrer">
```

**Review by jakxnz — CHANGES_REQUESTED** [review]
Looking good man. Codes getting better by the feature!

**Requirements**

A global footer with navigation *️⃣

Navigation *️⃣
image

We've already talk about an inbound nav here too.

Copyright ✅

image

File: src/js/components/config/team-config.js (hunk 18 above)
Comment (jakxnz) [reply src/js/components/config/team-config.js:10]: Also, is the 60px size enough for retina? Or do we want to go for 120 to be safe?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js (hunk 3 above)
Comment (jakxnz) [reply src/js/components/UnderflipTeam/UnderFlipTeam.js]: Aha! If you like 😂 Doesn't bother me

File: src/js/components/Footer/TheFooter.js (hunk 4 above)
Comment (jakxnz) [reply src/js/components/Footer/TheFooter.js]: Is it?

File: src/js/components/Footer/Footer.js [hunk 21]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { CFooter, CImg } from '@coreui/react';
+import UnderFlipTeam from '../UnderflipTeam/UnderFlipTeam';
+
+const Footer = () => {
+  const productUrl = 'https://shredindex.com';
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/Footer/Footer.js:6]: cool

File: src/js/components/UnderflipTeam/UnderFlipTeam.js (hunk 16 above)
Comment (Krank3n) [reply src/js/components/UnderflipTeam/UnderFlipTeam.js:5]: nice

File: src/js/components/Footer/Footer.js [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+const Footer = () => (
+  <CFooter fixed={false} className="p-4 h-auto footers-banner ">
+    <div className="m-auto">
+      <a className="mr-1" href={ProductOverview.shredIndex.url} target="_blank" rel="noopener">{ProductOverview.shredIndex.name}</a>
+      <span className="mr-1">
+        &copy;
+        {new Date().getFullYear()}
+      </span>
+      <span className="mr-1">Developed by</span>
+      <a href={ProductOverview.underFlip.url} target="_blank" rel="noopener ">{ProductOverview.underFlip.name}</a>
```
Comment (jakxnz) [inline src/js/components/Footer/Footer.js:15]: Can we use destructuring here?

File: src/js/components/FooterNav/FooterNav.js [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+          </CNav>
+        </CCardBody>
+      </CCard>
+    </CCol>
+  </CRow>
+);
+
+export default FooterNav;
+
+const FooterNavList = () => NavConfig.map((item) => (
```
Comment (jakxnz) [inline src/js/components/FooterNav/FooterNav.js:28]: Nice.

File: src/js/components/FooterNav/FooterNav.js [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+          <CNav className="justify-content-center">
+            <FooterNavList />
+          </CNav>
+        </CCardBody>
+      </CCard>
+    </CCol>
+  </CRow>
+);
+
+export default FooterNav;
```
Comment (jakxnz) [inline src/js/components/FooterNav/FooterNav.js:26]: Preferably put exports at the bottom of the document, so they're easy to reference

File: src/js/components/SupportBanner/SupportBanner.js [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+const SupportBanner = () => (
+  <div className="support-banner p-0 h-auto">
+    <div className="supporters container w-100 p-4 align-items-center d-flex flex-wrap">
+      <SupportItems />
+    </div>
+  </div>
+);
+
+export const SupportItems = () => SupportService
+  .map((item) => (
```
Comment (jakxnz) [inline src/js/components/SupportBanner/SupportBanner.js:14]:
```suggestion
export const SupportItems = () => SupportService.map((item) => (
```

Is there a reason for dropping the line?

File: src/js/components/FooterNav/FooterNav.js (hunk 23 above)
Comment (jakxnz) [inline src/js/components/FooterNav/FooterNav.js:28]: `item` is a little vague to me. If you keep using this everywhere it'll lose all meaning. Can this variable be more declarative?

File: src/js/components/TeamMember/TeamMember.js [hunk 26]
```diff
@@ -0,0 +1,25 @@
+import { CImg } from '@coreui/react';
+import React from 'react';
+
+const TeamMember = (props) => (
+  <div className="m-auto p-4 d-flex align-items-center">
+    <div>
+      <CImg
+        src={props.member.imgSrc}
```
Comment (jakxnz) [inline src/js/components/TeamMember/TeamMember.js:8]: Can you use destructuring for `props.member`? (Don't our linter rules prefer destructuring)?

File: src/js/components/UnderflipTeam/UnderFlipTeam.js [hunk 27]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import TeamConfig from '../config/team-config';
+import TeamMember from '../TeamMember/TeamMember';
+
+const UnderflipTeam = () => TeamConfig
+  .map((item) => (
+    <TeamMember member={item} />
+  ));
```
Comment (jakxnz) [inline src/js/components/UnderflipTeam/UnderFlipTeam.js:8]:
```suggestion
const UnderflipTeam = () => TeamConfig.map((item) => <TeamMember member={item} />);
```
Any reason for making this so many lines?

File: src/js/components/config/supporter-config.js [hunk 28]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+const SupportService = [
```
Comment (jakxnz) [inline src/js/components/config/supporter-config.js:3]: This isn't a service pattern, so can this name be more declarative?

**Review by jakxnz — CHANGES_REQUESTED** [review]
Getting cleaner and cleaner man! Keep it coming. Added a few more questions

**Review by jakxnz — CHANGES_REQUESTED** [review]
Yeah dude! This code is looking super tidy. Man, such a huge improvement over the life of this PR.

I've pushed a commit to resolve the spelling linting issue you were having, and added a few questions and comments.

Also, do you think you could squash this before the next review? 15 commits is quite a few for a component this size 😅

**Review by Krank3n — COMMENTED** [review]
> I've pushed a commit to resolve the spelling linting issue you were having, and added a few questions and comments.

Thank you so much dude

File: .eslintrc.json [hunk 29]
```diff
@@ -4,10 +4,21 @@
...
+      }
+    },
     "extends": [
         "plugin:react/recommended",
         "plugin:cypress/recommended",
-        "airbnb"
+        "airbnb",
+        "plugin:import/errors",
+        "plugin:import/warnings",
+        "plugin:import/typescript"
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply .eslintrc.json:21]: Sorry this was me trying to fix the linting errors... I'll take it out

File: src/js/components/Footer/Footer.js [hunk 30]
```diff
@@ -0,0 +1,26 @@
...
+  const { shredIndex, underflip } = ProductOverview;
+  return (
+    <CFooter fixed={false} className="p-4 h-auto footers-banner ">
+      <div className="m-auto">
+        <a className="mr-1" href={shredIndex.url} target="_blank" rel="noreferrer">{shredIndex.name}</a>
+        <span className="mr-1">
+          &copy;
+          {new Date().getFullYear()}
+        </span>
+        <span className="mr-1">Developed by</span>
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/Footer/Footer.js:16]: done

File: src/js/components/SupportBanner/SupportBanner.js [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+  <div className="support-banner p-0 h-auto">
+    <div className="supporters container w-100 p-4 align-items-center d-flex flex-wrap">
+      <Supporters />
+    </div>
+  </div>
+);
+
+export const Supporters = () => SupportItems.map((support) => (
+  <div className="m-auto p-3" key={support.name}>
+    <a href={support.url} rel="noopener noreferrer">
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/js/components/SupportBanner/SupportBanner.js:15]: yea I'll add some

File: src/scss/_custom.scss [hunk 32]
```diff
@@ -15,3 +16,14 @@ button {
     box-shadow: none !important;
   }
 }
+
+.c-avatar-img.round{
+  border-radius: 100%;
+}
+
+.developer-avatar {
+  width: 70px;
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/scss/_custom.scss:25]: added

File: src/scss/_supporters.scss [hunk 33]
```diff
@@ -0,0 +1,11 @@
+.supporters {
+  //max-width: 1100px;
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/scss/_supporters.scss:2]: removed

File: src/scss/_custom.scss [hunk 34]
```diff
@@ -15,3 +16,14 @@ button {
...
+  border-color: $primary;
```
[the remark this answers is not in the document]
Comment (Krank3n) [reply src/scss/_custom.scss:26]: Maybe why do you ask?

[the remark this answers is not in the document]
Comment (jakxnz) [reply src/scss/_custom.scss:26]: As far as I'm aware, `var()` has better compatibility with browsers, and is easier to debug.

**Review by jakxnz — APPROVED** [review]
It's gooooooooooood! 🚦 🟢🟢🟢🟢🟢🟢🟢🟢🟢🟢

**Pull request merged**